wallet/src/main/scala/org/bitcoins/wallet/models/TransactionTable.scala

```diff
@@ -0,0 +1,92 @@
+package org.bitcoins.wallet.models
+
+import org.bitcoins.core.protocol.transaction.Transaction
+import slick.jdbc.SQLiteProfile.api._
+import slick.lifted.ProvenShape
+import org.bitcoins.db.TableAutoInc
+import org.bitcoins.core.crypto.DoubleSha256DigestBE
+import org.bitcoins.db.DbRowAutoInc
+import org.bitcoins.core.protocol.script.ScriptPubKey
+
+/**
+  * Database representation of transactions
+  * relevant to our wallet.
+  */
+sealed trait TransactionDb[T <: TransactionDb[_]] extends DbRowAutoInc[T] {
+  val transaction: Transaction
+  lazy val txid: DoubleSha256DigestBE = transaction.txIdBE
+  val confirmations: Int
+
+}
+
+/** Transactions our wallet has received */
+final case class IncomingTransaction(
+    transaction: Transaction,
+    scriptPubKey: ScriptPubKey,
```

Comment:
i think output index is better than spk here, and we can add a method that gives amount, scriptPubKey, etc in the body of the case class

Reply:
The problem there is that we need to create a foreign key on `scriptPubKey` into address table. I don't know how to do that without having the actual SPK as a field in the case class...

Reply:
```scala
require(transaction.outputs.exists(_.scriptPubKey == scriptPubKey))
```